src/app/work-item/work-item-iteration-association-modal/work-item-iteration-association-modal.component.html

```diff
@@ -18,25 +25,25 @@
             <strong>{{ workItem.attributes['system.title'] }}</strong> with
           </label>
 
-          <div class="dib" dropdown id="iteration-select-dropdown">
-              <button #dropdownButton class="btn btn-default" type="button" dropdownToggle>
-                  {{ workItem.relationships.iteration?.attributes?.name || "Select iteration..." }} <span class="caret"></span>
-              </button>
-              <ul class="test" role="menu" dropdownMenu>
-                <li *ngFor="let iteration of iterations;" role="presentation">
-                  <a role="menuitem" tabindex="-1" (click)="onChangeIteration(iteration)">
-                    {{ iteration.attributes.name }}
-                  </a>
-                </li>
-              </ul>
+          <div class="parent-iteration-container input-group dropdown open active">
+            <input id="parent-iteration" class="form-control " type="text" #iterationSearch (keyup)="filterIteration($event)"
+            autocomplete="off" [value]="selectedIterationName" (focus)="iterationSearchFocus();">
+            <ul class="iteration-list dropdown-menu dropdown-ul" *ngIf="showIterationDropdown" #iterationList>
+              <li *ngIf="!filteredIterations.length">No matches found.</li>
```

Comment:
Typeahead dropdown needs to be reset before closing.
1. Open the iteration association modal
2. In the type-ahead, enter a string which results in "No matches found"
3. Close the modal
4. Open the modal - the type-ahead shows No matches found.